monitor/health/http.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package health
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ChainSafe/ChainBridge/core"
+	msg "github.com/ChainSafe/ChainBridge/message"
+	log "github.com/ChainSafe/log15"
+)
+
+type httpMetricServer struct {
+	port         int
+	timeDelay    int
+	core         *core.Core
+	blockheights map[msg.ChainId]blockHeightInfo
+}
+
+type blockHeightInfo struct {
+	height      *big.Int
+	lastUpdated time.Time
+}
+
+type httpMetricOptions struct {
+	port      int
+	timeDelay int
+	core      *core.Core
+}
+
+type httpResponse struct {
+	Data  string
+	Error string
+}
+
+func newhttpMetricServer(opts httpMetricOptions) *httpMetricServer {
+	return &httpMetricServer{
+		port:         opts.port,
+		core:         opts.core,
+		timeDelay:    opts.timeDelay,
+		blockheights: make(map[msg.ChainId]blockHeightInfo),
+	}
+}
+
+// Start starts the http metrics server
+func (s httpMetricServer) Start() {
+	log.Info("Metrics server started", "port", s.port)
+
+	// Setup routes
+	http.HandleFunc("/health", s.healthStatus)
+
+	// Start http server
+	err := http.ListenAndServe(":"+strconv.Itoa(s.port), nil)
+
+	if err == http.ErrServerClosed {
+		log.Info("Server is shutting down", err)
```

Comment:
Logs are very vague without context. Please use local logger or be more verbose.